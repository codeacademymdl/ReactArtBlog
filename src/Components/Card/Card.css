<<<<<<< HEAD
.main{
    justify-self: center;
    display: flex;
    flex-direction: column;
    border: 1px solid #787878;
    max-width: 350px;
    margin-left:5%;
    margin-right:5%;
}

.backgroundImage{

    width: 350px;
    height: 300px;
    background-size: cover;
}

.imageInfo{

    display:flex;
    justify-content: space-between;
    font-size:10px;
    color:gray;
    padding:10px;

}

.imageDesc{
    font-size: 20px;
    color:black;
    display:flex;
    flex-direction: column;
    padding:0px 10px;
}

.imageDesc span{
    margin-top:1em;;
}
.imageContent{
    font-size:0.5em;
    color:grey;
}

.imageFooter{
    display:flex;
    justify-content: space-between;
    color:black;
   
}


.reaction{
    display:flex;
}

.reaction img{
    width:40px;
    height:30px;
}
.imageFooter img{
    width:40px;
    height:30px;
}
=======
  .date {
    font-size: 10px;
  }
  
  .title {
    font-size: 18px;
  }
  
  .abstract {
    font-size: 10px;
  }

  .card {
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    padding: 8px;
    text-align: center;
    /* background-color: #f1f1f1; */
    background-color: rgb(255, 255, 255);
    margin-left: 5%;
    margin-right: 5%;
    margin-top: 2%;
    width: 200px;
  }
  
  .container {
    text-align: left;
    font-family: Arial, Helvetica, sans-serif;
  }
  
  .clap {
    width: 10%;
  }
  
  .heart {
    width: 10%; 
    float: right;
  }
  
  .image {
    width: 100%;
  }
  
  #readTime {
    float: right;
  }
  /*
  #readTime {
    float: left;
  } */
>>>>>>> 6be48bffc7bbbf05eda7ef6d565bf7d24ba2651e
