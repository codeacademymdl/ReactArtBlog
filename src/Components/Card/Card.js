import React, { Component } from 'react';
import './Card.css';
import BlackHeartImage from '../../Icons/heart-black.svg'
import RedHeartImage from '../../Icons/heart-red.svg';
import ClappingImage from '../../Icons/clapping.svg';


class Card extends Component {
    render() {
        return (
            <div className="main" >
                <div className="backgroundImage" style={{ backgroundImage: `url(${this.props.imageSrc})` }} />

                <div className="imageInfo">

                    <span>{this.props.date}</span><span>{this.props.readingTime}</span>
                </div>
                <div className="imageDesc">
                    <span >{this.props.title}</span>
                    <span className="imageContent">{this.props.description}</span>

                </div>
                <div className="imageFooter">
                    <div className="reaction">
                        <img src={ClappingImage} />
                        <span>{this.props.claps}</span>
                    </div>
                    {
                        this.props.liked?<img src={RedHeartImage} /> :<img src={BlackHeartImage} />   
                    } 

                </div>
            </div>
        );
    }
}

export default Card;
