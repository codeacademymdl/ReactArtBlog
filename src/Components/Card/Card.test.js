import React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Card from './Card'

const card = {
    "date": "2nd Januray, 2018",
    "readingTime": "2 mins",
    "title": "The future of abstract art and the culture ...",
    "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    "claps": 10,
    "liked": false,
    "image": "abstract.png"
}
describe('Card', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<Card />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});

describe('Card Counter', () => {
    it('checks if counter function increment by 1 ', () => {
        const wrapper = shallow(<Card
            date={card.date}
            readingTime={card.readingTime}
            title={card.title}
            description={card.description}
            claps={card.claps}
            liked={card.liked}
            imageSrc={`./Images/${card.image}`}
            
        />)
        expect(wrapper.instance().state.count).toEqual(card.claps);
        wrapper.instance().Counter();
        expect(wrapper.instance().state.count).toEqual(card.claps + 1);

    });

});

describe('Card likeDislike', () => {
    it('checks if state of islike is changed ', () => {
        const wrapper = shallow(<Card
            date={card.date}
            readingTime={card.readingTime}
            title={card.title}
            description={card.description}
            claps={card.claps}
            liked={card.liked}
            imageSrc={`./Images/${card.image}`}
            
        />)
        expect(wrapper.instance().state.isLike).toEqual(card.liked);
        wrapper.instance().likeDislike();
        expect(wrapper.instance().state.isLike).toEqual(!card.liked);

    });

});


