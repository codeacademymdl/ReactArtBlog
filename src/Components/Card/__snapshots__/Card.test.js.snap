// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  class="card"
>
  <img
    alt="abstract"
    className="image"
  />
  <div
    className="container"
  >
    <span
      className="date"
    />
    <span
      className="date"
      id="readTime"
    />
    <p
      className="title"
    />
    <p
      className="abstract"
    />
    <hr />
    <img
      alt="clap"
      className="clap"
      onClick={[Function]}
      src="clapping.svg"
    />
    <span />
    <img
      alt=""
      className="heart"
      onClick={[Function]}
      src="heart-red.svg"
    />
  </div>
</div>
`;
